satpy/composites/__init__.py

```diff
@@ -1908,6 +1908,9 @@ def __call__(self, projectables, *args, **kwargs):
         data_in = projectables[0]
         mask_in = projectables[1]
 
+        mask_in = mask_in.squeeze(drop=True)
+        mask_in = mask_in.round()  # Make sure to have whole numbers in case of smearing from resampling
```

Comment:
The mask variable could be anything including floating point. We can't have this `round` here.

```suggestion
```

If resampling is turning your category/flag variables into floating point values then you'll need to change how you do resampling.

Reply:
OK.